v2/hack/certify/certify.sh

```diff
@@ -0,0 +1,164 @@
+      echo "oc describe clusterserviceversion $CSV -n $2"
+      exit 1
+    fi
+		sleep $SLEEP_SHORT
+	done
+}
+
+
+if [ -z ${PYXIS_API_KEY+x} ]; then echo "PYXIS_API_KEY is unset"; exit 1; fi
+
+
+
+# Install Subscription
+oc apply -f sub.yaml
+
+# Verify Subscriptions
+checksub openshift-pipelines-operator-rh openshift-operators
+
+# Switch to certification namespace
+oc delete ns $CERT_NAMESPACE --ignore-not-found
+oc adm new-project $CERT_NAMESPACE
+oc project $CERT_NAMESPACE
+
+oc create secret generic pyxis-api-secret --from-literal pyxis_api_key=$PYXIS_API_KEY
+
+# Create the kubeconfig used by the certification pipeline
+oc delete secret kubeconfig --ignore-not-found
+oc create secret generic kubeconfig --from-file=kubeconfig=$KUBECONFIG
+
+# Import redhat catalogs
+oc import-image certified-operator-index \
+  --from=registry.redhat.io/redhat/certified-operator-index \
+  --reference-policy local \
+  --scheduled \
+  --confirm \
+  --all
+
+oc import-image redhat-marketplace-index \
+  --from=registry.redhat.io/redhat/redhat-marketplace-index \
+  --reference-policy local \
+  --scheduled \
+  --confirm \
+  --all
+
+CWD=$(pwd)
+TMP_DIR=$(mktemp -d 2>/dev/null || mktemp -d -t 'cptmpdir')
+
+# Install the Certification Pipeline
+cd $TMP_DIR
+git clone https://github.com/redhat-openshift-ecosystem/operator-pipelines
+cd operator-pipelines
+oc apply -R -f ansible/roles/operator-pipeline/templates/openshift/pipelines
+oc apply -R -f ansible/roles/operator-pipeline/templates/openshift/tasks
+cd $CWD
+
+oc apply -f https://raw.githubusercontent.com/tonytcampbell/operator-pipelines/preflight-fixes/ansible/roles/operator-pipeline/templates/openshift/tasks/preflight.yml
+
+
+# Add bundle to the fork
+cd $TMP_DIR
+git clone git@github.com:redhat-marketplace/certified-operators-preprod.git
+
+cd $CWD
+cd ../../
+
+rm -rf $TMP_DIR/certified-operators-preprod/operators/redhat-marketplace-operator/$VERSION
+mkdir -p $TMP_DIR/certified-operators-preprod/operators/redhat-marketplace-operator/$VERSION
+cp -r bundle/manifests $TMP_DIR/certified-operators-preprod/operators/redhat-marketplace-operator/$VERSION/
+cp -r bundle/metadata $TMP_DIR/certified-operators-preprod/operators/redhat-marketplace-operator/$VERSION/
+
+# The operator service account should be ommited in the bundle
+# It will fail certification
+# The service account will be created by OLM
+# kustomize questionable capability to remove the service account
+rm -Rf  $TMP_DIR/certified-operators-preprod/operators/redhat-marketplace-operator/$VERSION/manifests/redhat-marketplace-operator_v1_serviceaccount.yaml
+
+echo "organization: redhat-marketplace" > $TMP_DIR/certified-operators-preprod/config.yaml
+echo "cert_project_id: 5f68c9457115dbd1183ccab6" > $TMP_DIR/certified-operators-preprod/operators/redhat-marketplace-operator/ci.yaml
+
+cd $TMP_DIR/certified-operators-preprod
+git add --all
+git commit -m $VERSION
+git push
```

Comment:
Can you use branches or it has to be a main commit&push?

Reply:
they actually just changed the repo in the instructions, and the default branch was stage, now it is main
we can set this to our choice of branch, the git_branch is specified in the tekton run
let me know how you want to handle it

Reply:
Let's create a branch named for the release off the version and pass that. It'll let us keep stuff separate. Maybe at some point we can trigger actions on merge. If that's too complicated to do right now we can do follow on. First priority is a release of 2.4.0